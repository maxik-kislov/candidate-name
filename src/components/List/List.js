import React from 'react';
import PropTypes from 'prop-types';
import Tweet from '../Tweet/Tweet';
import './List.scss';

const List = ({ tweets }) => (
  <div className="list">
    <ul className="list__list">
      {tweets.length <= 0 && <span className="list__loading">Loading...</span>}
      {tweets.map((tweet, i) => (
        <Tweet
          key={i}
          username={tweet.username}
          text={tweet.text}
          image={tweet.image}
        />
      ))}
    </ul>
  </div>
);

List.defaultProps = {
  tweets: [],
};

List.propTypes = {
  tweets: PropTypes.array.isRequired,
};

export default List;
