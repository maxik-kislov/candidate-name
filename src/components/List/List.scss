.list {
  &__list {
    margin: 0;
    padding: 0;
    list-style: none;
  }

  &__loading {
    padding: 8px 0px;
    display: inline-block;
    font-style: italic;
  }
}
