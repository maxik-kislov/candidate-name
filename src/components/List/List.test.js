import setupTests from '../../setupTests';
import React from 'react';
import { shallow } from 'enzyme';
import List from './List';

const render = props => shallow(<List {...props} />);

describe('List', () => {
  it('should contain an ul element with a role and a class .list__list', () => {
    const element = render().find('.list__list');
    expect(element.exists()).toBeTruthy();
    expect(element.name()).toEqual('ul');
  });

  it('should display a "Loading..." message if results are empty', () => {
    const props = { tweets: [] };
    const element = render(props).find('.list__loading');
    expect(element.exists()).toBeTruthy();
  });

  it('should render list items correctly when tweets array has items', () => {
    const props = {
      tweets: [
        {
          image:
            'https://s3.amazonaws.com/uifaces/faces/twitter/pixage/128.jpg',
          text:
            '10001. "It is straight enough. I should like to know who sold you the\ngeese which you supplied to the Alpha."',
          username: 'Ferne.Dickinson1',
        },
        {
          image:
            'https://s3.amazonaws.com/uifaces/faces/twitter/kazaky999/128.jpg',
          text:
            '10000. Godfrey\nNorton, as our client is to its coming to the eyes of his\nprincess.',
          username: 'Jena_Heller',
        },
      ],
    };

    const items = render(props).find('Tweet');
    expect(items.length).toEqual(props.tweets.length);

    const newItems = items.map(item => item.props());
    expect(newItems).toEqual(props.tweets);
  });

  it('should render correctly', () => {
    expect(render().getElements()).toMatchSnapshot();
  });
});
