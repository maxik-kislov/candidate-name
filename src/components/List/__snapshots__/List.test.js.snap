// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render correctly 1`] = `
Array [
  <div
    className="list"
  >
    <ul
      className="list__list"
    >
      <span
        className="list__loading"
      >
        Loading...
      </span>
    </ul>
  </div>,
]
`;
